web/js/link/wv.link.ui.js

```diff
@@ -68,6 +68,54 @@ wv.link.ui = wv.link.ui || function(models, config) {
                 "</div>";
         }
         item += "</div>";
+
```

Comment:
I would work on breaking up the entire self.show method into a compilation of smaller functions. Rough example:

```
self.show = function() {
    var icons, el;
    
    el = document.createElement('div'); 
    icons = createShareIcons(el);
}
var createShareIcons = function(someElem) {
    var fb, reddit;
 
    fb = createIcon('fb'. xxx1);
    reddit = createIcon('reddit', xxx2);
    someElement.appendChild(fb);
    someElement.appendChild(reddit)
    return someElement;
}
var createIcon = function(type, id){
    // create elements based on parameters
}
```

After looking at this again... I think it could be cool to build it in react. Let me know @ZachTRice if you would be interested in migrating this code to react. I could definitely help you get started with that if interested.